lib/src/main/java/com/orange/ods/compose/component/navigationdrawer/OdsModalDrawer.kt

```diff
@@ -0,0 +1,253 @@
+                        text = item.label
+                    )
+                }
+            }
+        }
+        is OdsModalDrawerListItem -> {
+            {
+                OdsListItem(
+                    modifier = Modifier
+                        .iconType(OdsListItemIconType.Icon)
+                        .background(color = Color.Red),
+                    text = item.text,
+                    icon = item.icon?.let { { OdsListItemIcon(painterResource(id = it)) } })
+            }
+        }
+        is OdsModalDrawerDivider -> {
+            {
+                OdsDivider()
+            }
+        }
+    }
+}
+
+data class OdsModalDrawerHeaderParametersProvider(
+    var modifier: Modifier = Modifier,
+    var title: String,
+    var imageContentDescription: String? = null,
+    var backgroundImage: Painter? = null,
+    var subtitle: String? = null,
+    var avatar: Painter? = null
+)
+
+@Composable
+internal fun OdsModalDrawerHeader(
+    headerContent: OdsModalDrawerHeaderParametersProvider
+) {
+    Box(
+        modifier = headerContent.modifier
+            .fillMaxWidth()
+            .height(
+                if (headerContent.backgroundImage != null || headerContent.avatar != null) 167.dp else if (headerContent.subtitle != null) dimensionResource(
+                    id = R.dimen.list_two_line_item_height
+                ) else dimensionResource(id = R.dimen.list_single_line_item_height)
+            ),
+    ) {
+        headerContent.backgroundImage?.let {
+            Image(
+                painter = it,
+                contentDescription = headerContent.imageContentDescription,
+                contentScale = ContentScale.Crop,
+                modifier = headerContent.modifier
+                    .fillMaxWidth()
+            )
+            Surface(
+                color = Color.Black.copy(alpha = 0.8f),
+                modifier = Modifier
+                    .align(Alignment.BottomStart)
+                    .fillMaxWidth()
+            ) {
+                OdsHeaderText(headerContent = headerContent, color = Color.White)
+            }
+        }
+        headerContent.avatar?.let {
+            OdsImageCircleShape(
+                painter = it,
+                Modifier.padding(start = dimensionResource(id = R.dimen.spacing_m), top = 40.dp)
+            )
+            Column(Modifier.align(Alignment.BottomStart)) {
+                OdsHeaderText(headerContent = headerContent, color = Color.Black)
+            }
+
+        }
+        if (headerContent.backgroundImage == null && headerContent.avatar == null) OdsHeaderText(headerContent = headerContent, color = Color.Black)
+    }
+}
+
+@Composable
+fun OdsHeaderText(headerContent: OdsModalDrawerHeaderParametersProvider, color: Color) {
+    Column(
+        Modifier
+            .padding(start = dimensionResource(id = R.dimen.spacing_m), top = 12.dp, bottom = 12.dp)
```

Comment:
The spacing `12dp` doesn't exist in the guideline. I think you have to create a constant to define the height of the header text and to center vertically texts inside this layout.